tests/common/include/tests/common/tools_propagator.inl

```diff
@@ -48,7 +53,118 @@ TEST(ALGEBRA_PLUGIN, propagator) {
     using detray_propagator = propagator<detray_stepper, detray_navigator>;
     detray_propagator p(std::move(s), std::move(n));
 
-    void_track_inspector vi;
+    void_propagator_inspector vi;
 
     /*auto end = */ p.propagate(traj, vi);
 }
+
+struct helix_inspector {
+
+    helix_inspector(const helix_gun& helix) : _helix(helix) {}
+
+    template <typename navigator_state_t, typename stepper_state_t>
+    DETRAY_HOST_DEVICE void operator()(const navigator_state_t& /*navigation*/,
+                                       const stepper_state_t& stepping) {
+        auto pos = stepping().pos();
+        auto true_pos = _helix(stepping._path_accumulated);
+
+        auto relative_error = 1 / stepping._path_accumulated * (pos - true_pos);
+
+        EXPECT_NEAR(getter::norm(relative_error), 0, epsilon);
+    }
+
+    helix_gun _helix;
+};
+
+struct print_inspector {
```

Comment:
I think, in the end, we should put some of these inspectors into their own header

Reply:
Their use cases are quite limited to the unit tests though. But I have no objections

Reply:
True, I thought about the ```common/tools``` folder in the tests